### PR TITLE
Crop params in percentage

### PR DESCRIPTION
**Pull request by pletessier**
Hi Adrian,
As discussed in #26, here is a proposal for image cropping using percentage of image dimensions.

### CODE REVIEW

File: imagorpath/params.go [hunk 1]
```diff
@@ -20,33 +20,37 @@ type Filters []Filter
...
+	Path              string  `json:"path,omitempty"`
+	Image             string  `json:"image,omitempty"`
+	Unsafe            bool    `json:"unsafe,omitempty"`
+	Hash              string  `json:"hash,omitempty"`
+	Meta              bool    `json:"meta,omitempty"`
+	Trim              bool    `json:"trim,omitempty"`
+	TrimBy            string  `json:"trim_by,omitempty"`
+	TrimTolerance     int     `json:"trim_tolerance,omitempty"`
+	CropLeft          int     `json:"crop_left,omitempty"`
+	CropLeftPercent   float64 `json:"crop_left_percent,omitempty"`
```
Comment (cshum) [inline imagorpath/params.go:33]: What do you think about e.g. CropLeft and CropLeftPercent to combine as 1 value

- value < 1: evaluate as percentage
- value >= 1: evaluate as pixel

seems viable?

**Review by cshum — COMMENTED** [review]
Thanks for your contributions! Please see my comments :)

File: imagorpath/params.go (hunk 1 above)
Comment (cshum) [reply imagorpath/params.go:33]: Though we will then need to deal with floating points more carefully, when evaluating pixels.

Comment (pletessier) [reply imagorpath/params.go:33]: Seems viable to me. I can try it and see if floating points is not messing up anything.

File: imagorpath/generate.go [hunk 2]
```diff
@@ -26,6 +26,10 @@ func generate(p Params) string {
 		parts = append(parts, fmt.Sprintf(
 			"%dx%d:%dx%d", p.CropLeft, p.CropTop, p.CropRight, p.CropBottom))
 	}
+	if p.CropTopPercent > 0 || p.CropRightPercent > 0 || p.CropLeftPercent > 0 || p.CropBottomPercent > 0 {
```
[the remark this answers is not in the document]
Comment (pletessier) [reply imagorpath/generate.go:29]: Yes, totally agree !

File: imagorpath/generate.go [hunk 3]
```diff
@@ -26,6 +26,10 @@ func generate(p Params) string {
...
+		parts = append(parts, fmt.Sprintf(
+			"%.6fx%.6f:%.6fx%.6f", p.CropLeftPercent, p.CropTopPercent, p.CropRightPercent, p.CropBottomPercent))
```
[the remark this answers is not in the document]
Comment (pletessier) [reply imagorpath/generate.go:31]: I think it's sufficient for the test. Anywhere else, we need to keep enough precision to deal with images up to 10000x10000, so "%.4f" would be enough.

[the remark this answers is not in the document]
Comment (cshum) [reply imagorpath/generate.go:31]: what about formatting to its nearest precision, i.e.
```go
strconv.FormatFloat(10.009, 'f', -1, 64)
// "10.009"
```

**Pull request merged**